### DISCUSSION TITLE
Not Safe For Work images in my email

### DISCUSSION THREAD

**Discussion by AnthonyB28** [community/community#17933] [Projects and Issues]
Both my personal and professional work emails were in a mailing list on a repository. This repository is a big professional one, which was attacked by malicious people through pull requests which spammed nearly half a million people on GitHub in this repository.

After 3 hours of this and locking the topic and the emails being caught up, someone did it again but this time people responding to the PR included NSFW images in the comments.

This image loads in email software such as Gmail automatically. So now I have NSFW images embedded and loading in my work and personal devices that I did not need or want to see. Even if moderation cleans it up, the email notifications still went out. So I don't feel it is fair the issue is on them, to me the issue seems to be on GitHub for embedding such images into emails automatically. This should not be the case in my opinion, especially if it can be so easily abused.

Prevent auto embedded images in email notifications for comments on a PR please, or a better alternative to detect and prevent such attacks automatically

**Comment by kingsapo** [comment]
Totally agree with this. I feel like we're talking about the same thing here, but for reference, here's the repo I was affected by: https://github.com/EpicGames/Signup

They fixed the spam issue by automatically closing all pull requests, which isn't exactly a great solution long-term, but that's separate from the NSFW issue. I received a pretty gross meme image of a guy's butt in my email today from a now-deleted comment on one of the PRs on this repo. It's great that the comment was deleted, but the image is still on GitHub's servers, and still shows up in my email even now.

One solution is to implement content scanning like the other social media sites have, but I recognize the resources required to do so. A quicker solution could be to limit image embedding to people with more than x number of comments. macrumors.com does something similar to this when they post about political issues to prevent spam, like [this](https://www.macrumors.com/2022/06/02/apple-retail-changes-amid-unionization-efforts/) post for example.

**Comment by 2rs2ts** [comment]
An easy-to-implement solution would be for users to have an option to disable embedded images and other rich media in the emails they receive through the notification settings for their account. If that setting is disabled, the email templates should render some placeholder that just shows an indication that there is media there, along with any alt text, if present.
